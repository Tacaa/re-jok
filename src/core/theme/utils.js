import {
  defaultTheme
} from './index';


export const configTheme = (theme: Object) => {
  let jokTheme = {...defaultTheme, ...theme};
  return jokTheme;
}
