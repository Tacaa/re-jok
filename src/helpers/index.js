export const generalId = () => {
  return '_' + Math.random().toString(36).substr(2, 9);
}

//Default Function
export const noop = () => {}
