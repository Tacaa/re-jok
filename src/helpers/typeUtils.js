/** Check element has child or not**/
export const isChild = children => children === null
    || children === undefined
    || (Array.isArray(children) && children.length === 0)

/** Check undefined**/

export const isUndef = value => typeof value === 'undefined';
/** Check function or not**/

export const isFunction = value => !isUndef(value) && typeof value === 'function';

/** Check Object or not**/
export const isObject = value =>
      typeof value === 'object' && value !== null && !Array.isArray(value);

/** Check Array or not**/
export const isArray = value => Array.isArray(value);

/** Check Number or not**/
export const isNumber = value => Number.isInteger(value) && typeof value === "number"
