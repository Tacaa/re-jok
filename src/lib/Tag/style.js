import styled,{css} from 'styled-components';
import { shadeColor } from '../globals';
import Icon from '../Icon';


export const StyledIconClose = styled(Icon)`
 text-align: center;
 position: relative;
 cursor: pointer;
 height: 1.6rem;
 width: 0;
 line-height: 1.6rem;
 vertical-align: middle;
 top: -.1rem;
 right: -.5rem;
 opacity: 0;
 transition: all .2s;

`
export const StyledTag = styled.span`
  display: inline-flex;
  height: ${props => props.theme.size[props.size].height};
  line-height: ${props => props.theme.size[props.size].height};
  justify-content: center;
  align-items: center;
  flex-wrap: wrap;

  padding: 0 1.1rem;
  margin-right: .8rem;
  margin-bottom: .8rem;
  background-color: ${ ({theme, color}) => color ? color : shadeColor(theme.background.paper, -5)};
  border: .1rem solid transparent;
  border-radius: .2rem;
  color: ${ ({theme, color}) => color ? theme.text.default : theme.text.secondary};
  cursor: pointer;
  font-size: ${props => props.theme.size[props.size].fontSize};
  position: relative;
  transition: all .2s;

  &:hover{
    color: ${ ({theme, color}) => color ? theme.text.default : theme.text.regular};
    ${
      props => props.closable && css`
         padding-right: 1.4rem;
         & ${StyledIconClose}{
           width: 1.6rem;
           opacity: 1;
           color: ${ ({color, theme}) => color && theme.text.default};
         }
      `
    }
  }
`
