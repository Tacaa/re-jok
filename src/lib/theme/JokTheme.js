// @flow
import * as React from 'react';
import { ThemeProvider } from 'styled-components';
import { defaultTheme } from './index';

type Props = {
  /**Theme of project**/
  theme: Object,
  /**Children can be anything**/
  children: any,
}

const defaultProps = {
  theme: {}
}

class JokTheme extends React.Component<Props>{
  static defaultProps= defaultProps;

  checkDefaultTheme = ()=>{
    const {theme} = this.props;
    if(Object.keys(theme).length === 0 && theme.constructor === Object) return true;
    return false;
  }

  render(){
    const {
      children,
      theme
    } = this.props;

    let useDefaultTheme = this.checkDefaultTheme();

    return (
      <ThemeProvider theme={useDefaultTheme? defaultTheme : theme}>
        {children}
      </ThemeProvider>
    )
  }
}

export default JokTheme
