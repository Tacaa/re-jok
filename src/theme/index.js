// @flow
// export our theme to the provider
// This is globally available in styled-components when interpolating a function like so:
// ${(props) => props.theme}
// Or using import { withTheme } from 'styled-components';


export const defaultTheme = {
  size: {
    small: {
      fontSize: '1.2rem',
      padding: '0.5rem 0.9rem',
      "lineHeight": "1",
      "borderRadius": ".2rem",
      width: "2.4rem",
      height: "2.4rem"
    },
    default: {
      fontSize: '1.4rem',
      padding: '0.9rem 1.8rem',
      lineHeight: "1.5",
      borderRadius: ".4rem",
      width: "3.2rem",
      height: "3.2rem"
    },
    medium: {
      fontSize: '1.8rem',
      padding: '1.4rem 2.7rem',
      lineHeight: "1.5",
      borderRadius: ".6rem",
      width: "4rem",
      height: "4rem"

    },
    large: {
      fontSize: '2rem',
      padding: '1.8rem 3.6rem',
      lineHeight: "2",
      borderRadius: ".6rem",
      width: "4.8rem",
      height: "4.8rem"
    }
  },
  grid:{
    xs: 768,
    sm: 768,
    md: 992,
    lg: 1200,
    xl: 1920
  },
  text: {
    primary: "rgba(0, 0, 0, 0.85)",
    secondary: "rgba(0, 0, 0, 0.48)",
    hint: "rgba(0, 0, 0, 0.38)"
  },
  background: {
    paper: "#fff",
    default: "#fafafa"
  },
  typography: {
    title: {
      fontSize: "2rem",
      fontWeight: "600",
      lineHeight: "2.8rem"

    },
    subtitle: {
      fontSize: "1.6rem",
      fontWeight: "400",
      lineHeight: "2.4rem"
    }
  },
  primary: {
    main: "#1c8ffb",

  },
  secondary: {
    main: "#1b1c1d",
  },
  info: {
    main: "#1890ff"
  },
  error: {
    main: "#f5222d"
  },
  success: {
    main: "#52c41a"
  },
  warning: {
    main: "#faad14"
  },
  zIndex: {
    base: 1,
    background: 0,
    hidden: -1,
    card: 2,
    loading: 3,
    avatar: 3,
    form: 3,
    search: 3,
    badge: 100,
    modal: 1500,
  }
};

const changeTheme = (defaultTheme, theme)=>{
  let cloneTheme = {...defaultTheme};


  Object.keys(theme).forEach(function (key) {
    if ( 'object' === typeof theme[key] && !Array.isArray(theme[key]) ) {
    	changeTheme(cloneTheme[key], theme[key])
    }
    else if ( 'string' === typeof theme[key] || 'number' === typeof theme[key] ) {
      cloneTheme[key] = theme[key]
    }
  })
  return cloneTheme;
}

export const setTheme = (theme: Object) => {

   const newTheme = changeTheme(defaultTheme, theme);

   return newTheme;

}
