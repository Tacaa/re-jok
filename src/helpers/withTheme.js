import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';
import {defaultTheme} from '../theme';
import { ThemeProvider } from 'styled-components';
export function renderWithTheme(component) {

  return renderer.create(
    <ThemeProvider theme={defaultTheme}>
      {component}
    </ThemeProvider>
  );
};

export const shallowWithTheme = (tree, theme) => {
  const context = shallow(<ThemeProvider theme={theme} />)
    .instance()
    .getChildContext()
  return shallow(tree, { context })
};

export const mountWithTheme = (tree, theme) => {
  const context = shallow(<ThemeProvider theme={theme} />)
    .instance()
    .getChildContext()

  return mount(tree, {
    context,
    childContextTypes: ThemeProvider.childContextTypes
  })
};
