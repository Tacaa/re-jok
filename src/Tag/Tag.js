// @flow
import * as React from 'react';
import {
  StyledTag,
  StyledIconClose,
} from './style';



import { isFunction } from '../helpers/typeUtils';
import { noop } from '../helpers';



type Props = {
  /** Style of Tag Component**/
  style?: Object,
  /** Class of Tag Component**/
  className?: string,
  /** Children of Tag Component could be anything**/
  children?: any,
  /** If you want to add more tag.. you should use tag key**/
  tagKey?: string | number,
  /** Set color of Tag **/
  color?: string,
  /** Set tag is closable or not**/
  closable: boolean,
  /** Set size of Tag**/
  size: 'small' | 'default' | 'medium' | 'large',
  /** Callback function when close Tag... it must be work with closable**/
  onClose: Function,
}

const defaultProps = {
  closable: false,
  size: 'default',
  onClose: noop
}


class Tag extends React.Component<Props>{

  static defaultProps = defaultProps;

  handleClose = (e: SyntheticEvent<HTMLElement>) => {
    const { onClose } = this.props;

   isFunction(onClose) && onClose(e, {...this.props});
  }

  render(){
    const {
      children,
      closable,
      ...rest
    } =  this.props;
    return (
      <StyledTag closable={closable} {...rest}>
        {children}
        { closable && <StyledIconClose onClick={this.handleClose} name="times"/> }
      </StyledTag>
    )
  }
}

export default Tag
