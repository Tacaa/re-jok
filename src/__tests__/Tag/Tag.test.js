import React from 'react';
import Tag from '../../Tag/Tag';


import {defaultTheme} from '../../theme';

import {shallowWithTheme, mountWithTheme, renderWithTheme} from '../../helpers/withTheme';
import 'jest-enzyme';
import 'jest-styled-components';

describe('<Tag />', () => {
  it('should render', () => {
    const tree = renderWithTheme(
      <Tag>Tag 1</Tag>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render with close icon', () => {
    const tree = renderWithTheme(
      <Tag closable>Tag 1</Tag>
    ).toJSON();
    expect(tree).toMatchSnapshot();


  });

  it('should render with size', () => {
    const tree = renderWithTheme(
      <Tag size="medium">Tag 3</Tag>

    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render with color', () => {
    const tree = renderWithTheme(
      <Tag color="#9c27b0">Tag 3</Tag>

    ).toJSON();
    expect(tree).toMatchSnapshot();


  });





})
