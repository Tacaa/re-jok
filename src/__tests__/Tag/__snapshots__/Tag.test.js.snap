// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tag /> should render 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  height: 3.2rem;
  line-height: 3.2rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  padding: 0 1.1rem;
  margin-right: .8rem;
  margin-bottom: .8rem;
  background-color: #ededed;
  border: .1rem solid transparent;
  border-radius: .2rem;
  color: rgba(0,0,0,0.48);
  cursor: pointer;
  font-size: 1.4rem;
  position: relative;
  -webkit-transition: all .2s;
  transition: all .2s;
}

.c0:hover {
  color: rgba(0,0,0,.64);
}

<span
  className="c0"
  size="default"
>
  Tag 1
</span>
`;

exports[`<Tag /> should render with close icon 1`] = `
.c3 {
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
  display: inline-block;
}

.c2 {
  text-align: center;
  position: relative;
  cursor: pointer;
  height: 1.6rem;
  width: 0;
  line-height: 1.6rem;
  vertical-align: middle;
  top: -.1rem;
  right: -.5rem;
  opacity: 0;
  -webkit-transition: all .2s;
  transition: all .2s;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  height: 3.2rem;
  line-height: 3.2rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  padding: 0 1.1rem;
  margin-right: .8rem;
  margin-bottom: .8rem;
  background-color: #ededed;
  border: .1rem solid transparent;
  border-radius: .2rem;
  color: rgba(0,0,0,0.48);
  cursor: pointer;
  font-size: 1.4rem;
  position: relative;
  -webkit-transition: all .2s;
  transition: all .2s;
}

.c0:hover {
  color: rgba(0,0,0,.64);
  padding-right: 14px;
}

.c0:hover .c1 {
  width: 1.6rem;
  opacity: 1;
}

<span
  className="c0"
  size="default"
>
  Tag 1
  <div
    className="c1 c2 c3"
    color={undefined}
    onClick={[Function]}
    style={undefined}
  >
    <span
      aria-hidden={true}
      className="fa fa-times"
    />
  </div>
</span>
`;

exports[`<Tag /> should render with color 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  height: 3.2rem;
  line-height: 3.2rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  padding: 0 1.1rem;
  margin-right: .8rem;
  margin-bottom: .8rem;
  background-color: #9c27b0;
  border: .1rem solid transparent;
  border-radius: .2rem;
  color: #fff;
  cursor: pointer;
  font-size: 1.4rem;
  position: relative;
  -webkit-transition: all .2s;
  transition: all .2s;
}

.c0:hover {
  color: #fff;
}

<span
  className="c0"
  color="#9c27b0"
  size="default"
>
  Tag 3
</span>
`;

exports[`<Tag /> should render with size 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  height: 4rem;
  line-height: 4rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  padding: 0 1.1rem;
  margin-right: .8rem;
  margin-bottom: .8rem;
  background-color: #ededed;
  border: .1rem solid transparent;
  border-radius: .2rem;
  color: rgba(0,0,0,0.48);
  cursor: pointer;
  font-size: 1.8rem;
  position: relative;
  -webkit-transition: all .2s;
  transition: all .2s;
}

.c0:hover {
  color: rgba(0,0,0,.64);
}

<span
  className="c0"
  size="medium"
>
  Tag 3
</span>
`;
